SteamKit2/SteamKit2/Types/SteamID.cs

```diff
@@ -459,10 +459,10 @@ public bool IsValid
         {
             get
             {
-                if ( this.AccountType <= EAccountType.Invalid || this.AccountType >= EAccountType.Max )
+                if ( this.AccountType <= EAccountType.Invalid || this.AccountType > EAccountType.AnonUser )

```

Comment:
This is prone to errors in the future, especially if the enum gets extended, it's probably better to use a combination of `!Enum.IsDefined() || this.AccountType == EAccountType.Invalid` or if you don't want the performance cost with it, explicitly extract it to some place that tracks valid account types.

Same below.

Reply:
If either of these enums change, SteamID code will require changes to handle them regardless.

Reply:
Can we add a unit test then that these values are the highest values, with a comment to update the SteamID code if it ever changes?

Reply:
Done.